code/modules/mob/living/basic/guardian/guardian_types/explosive.dm

```diff
@@ -0,0 +1,77 @@
+/// A durable guardian which can convert objects into hidden explosives.
+/mob/living/basic/guardian/explosive
+	guardian_type = GUARDIAN_EXPLOSIVE
+	melee_damage_lower = 15
+	melee_damage_upper = 15
+	damage_coeff = list(BRUTE = 0.6, BURN = 0.6, TOX = 0.6, CLONE = 0.6, STAMINA = 0, OXY = 0.6)
+	range = 13
+	playstyle_string = span_holoparasite("As an <b>explosive</b> type, you have moderate close combat abilities and are capable of converting nearby items and objects into disguised bombs via right-click.")
+	creator_name = "Explosive"
+	creator_desc = "High damage resist and medium power attack. Can turn any object, including objects too large to pick up, into a bomb, dealing explosive damage to the next person to touch it. The object will return to normal after the trap is triggered or after a delay."
+	creator_icon = "explosive"
+	/// Ability which plants bombs
+	var/datum/action/cooldown/mob_cooldown/explosive_booby_trap/bomb
+
+/mob/living/basic/guardian/explosive/Initialize(mapload, datum/guardian_fluff/theme)
+	. = ..()
+	bomb = new(src)
+	bomb.Grant(src)
+
+/mob/living/basic/guardian/explosive/Destroy()
+	QDEL_NULL(bomb)
```

Comment:
```suggestion
	bomb = null
```

datum action will handle cleaning itself up due to the `COMSIG_QDELING` signal it has registered, lets just clean up the hanging ref and move on

Reply:
I prefer leaving the qdel null to be clear what's going on, it's a no-op anyways.